src/ioctl.rs

```diff
@@ -182,12 +348,68 @@ pub unsafe fn ioctl_with_mut_ref<F: AsRawFd, T>(fd: &F, req: c_ulong, arg: &mut
     )
 }
 
-/// Run an ioctl with a raw pointer.
+/// A wrapper of unsafe [libc::ioctl] with a raw pointer.
+///
+/// # Arguments
+///
+/// * `fd`: an open file descriptor.
+/// * `req`: a device-dependent request code.
+/// * `arg`: a raw pointer passed to ioctl.
+///
+/// # Safety
+///
+/// The caller should ensure to pass a valid file descriptor and have the
+/// return value checked.
+///
+/// # Examples
+///
+/// ```
+/// extern crate vmm_sys_util;
+/// #
+/// # use std::fs::File;
+/// # use std::os::raw::c_ulong;
+/// # use std::os::unix::io::FromRawFd;
+/// use vmm_sys_util::ioctl::ioctl_with_ptr;
+///
+/// const TUNSETOFFLOAD: c_ulong = 0x4004_54d0;
+/// let bytes : Vec<u8> = vec![0; 10];
+/// let file = unsafe { File::from_raw_fd(0) };
+///
+/// let ret = unsafe { ioctl_with_ptr(&file, TUNSETOFFLOAD, bytes.as_ptr()) };
+/// ```
 pub unsafe fn ioctl_with_ptr<F: AsRawFd, T>(fd: &F, req: c_ulong, arg: *const T) -> c_int {
     libc::ioctl(fd.as_raw_fd(), req as IoctlRequest, arg as *const c_void)
 }
 
-/// Run an ioctl with a mutable raw pointer.
+/// A wrapper of unsafe [libc::ioctl] with a mutable raw pointer.
```

Comment:
```[libc::ioctl]``` is an empty link. I think we should keep the previous documentation: ``` Run an ioctl with a mutable raw pointer.```

Reply:
This stands for all the other comments for the ioctl wrappers.

Reply:
[libc::ioctl] is automatically linked to the right place when we kick it as I saw. 
So we would like not point out what ioctl actually is here?

Reply:
When I run `cargo doc --open` the `[libc::ioctl]` does not point to anything. Furthermore, it is printed as `[libc::ioctl]`. I think you can just leave it as it was before or add a link to http://man7.org/linux/man-pages/man2/ioctl.2.html

Reply:
Strange... Sure, let's just keep it as before.

Reply:
@liujing2 have you run `cargo doc --open`?

Reply:
yes, of course. I can kick the link in my side.